httpconnect/httpconnect_test.go

```diff
@@ -0,0 +1,53 @@
+package httpconnect
+
+import (
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/getlantern/testify/assert"
+)
+
+func TestFilterTunnelPorts(t *testing.T) {
+	filter, _ := New(http.NotFoundHandler(), AllowedPorts([]int{443, 8080}))
+	server := httptest.NewServer(filter)
+	defer server.Close()
+	u, _ := url.Parse(server.URL)
+	client := http.Client{Transport: &http.Transport{
+		Dial: func(network, addr string) (net.Conn, error) {
+			return net.Dial("tcp", u.Host)
+		},
+	}}
+
+	req, _ := http.NewRequest("CONNECT", "http://site.com", nil)
+	resp, _ := client.Do(req)
+	_ = resp.Body.Close()
+	assert.Equal(t, http.StatusBadRequest, resp.StatusCode, "")
```

Comment:
In these tests, it would be helpful to fill out the description of the test condition, both to help the reader understand the code and to help the tester understand a test failure if it happens.